cmd/traffic/cmd/agent/fwdstate.go

```diff
@@ -94,48 +111,217 @@ func (pm *ProviderMux) CreateClientStream(ctx context.Context, tag tunnel.Tag, s
 	return pm.AgentProvider.CreateClientStream(ctx, tag, sessionID, id, roundTripLatency, dialTimeout)
 }
 
-// interceptSpecsConflict determines if two intercept specs would conflict with each other.
-// Two specs conflict if they would route the same traffic to different destinations.
-func interceptSpecsConflict(spec1, spec2 *manager.InterceptSpec) bool {
-	// Fast path: For TCP intercepts (no filters), they always conflict on the same port
-	if len(spec1.HeaderFilters) == 0 && len(spec1.PathFilters) == 0 &&
-		len(spec2.HeaderFilters) == 0 && len(spec2.PathFilters) == 0 {
-		return true
+// normalizeHeaderFilters returns a new map with all header keys normalized to lowercase.
+// HTTP headers are case-insensitive per RFC 7230.
+func normalizeHeaderFilters(headers map[string]string) map[string]string {
+	if len(headers) == 0 {
+		return headers
+	}
+	normalized := make(map[string]string, len(headers))
+	for key, value := range headers {
+		normalized[strings.ToLower(key)] = value
```

Comment:
An alternative to using `strings.ToLower` here, which I think produces more readable headers once they are displayed, is to use the `textproto.CanonicalMIMEHeaderKey`.